flow/core/rewards.py

```diff
@@ -441,3 +441,52 @@ def miles_per_gallon(env, veh_ids=None, gain=.001):
     mpg /= 1609.0
 
     return mpg * gain
+
+
+def instantaneous_power(env, veh_ids=None, gain=.001):
+    """Calculate the instantaneous power for every simulation step specific to the vehicle type.
+
+    The energy model used is based on the vehicle type:
+    veh_id = "Prius"  --> energy model is PriusEnergy in flow.core.kernel.toyata_energy
+    veh_id = "Tacoma" --> energy model is TacomaEnergy in flow.core.kernel.toyata_energy
+    veh_id = "EV"     --> energy model is PDMElectric in flow.core.kernel.power_demand
+    veh_id = "human"  --> energy model is PDMCombustionEngine in flow.core.kernel.power_demand
+
+    Parameters
+    ----------
+    env : flow.envs.Env
+        the environment variable, which contains information on the current
+        state of the system.
+    veh_ids : [list]
+        list of veh_ids to compute the reward over
+    gain : float
+        scaling factor for the reward
+    """
+
+    if veh_ids is None:
+        veh_ids = env.k.vehicle.get_ids()
+    elif not isinstance(veh_ids, list):
+        veh_ids = [veh_ids]
+    for veh_id in veh_ids:
+        speed = env.k.vehicle.get_speed(veh_id)
+        prev_speed = env.k.vehicle.get_previous_speed(veh_id)
+        # negative acceleration indicates regenerative breaking for some energy models
+        accel = (speed - prev_speed) / env.sim_step
```

Comment:
for accel, you can use:
`env.k.vehicle.get_accel_no_noise_with_failsafe(veh_id)`
then you don't need prev_speed anymore